junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -86,6 +102,201 @@ fun assertAll(vararg executables: () -> Unit) =
 fun assertAll(heading: String?, vararg executables: () -> Unit) =
     assertAll(heading, executables.toList().stream())
 
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: String? = ...
+ *
+ * assertNull(string)
+ *
+ * // compiler won't allow even safe calls, since the string is always null
+ * // string?.isNotEmpty()
+ * ```
+ * @see Assertions.assertNull
+ */
+@API(since = "5.11", status = EXPERIMENTAL)
+fun assertNull(actual: Any?) {
+    contract {
+        returns() implies (actual == null)
+    }
+
+    Assertions.assertNull(actual)
+}
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: String? = ...
+ *
+ * assertNull(string, "Should be nullable")
+ *
+ * // compiler won't allow even safe calls, since the string is always null
+ * // string?.isNotEmpty()
+ * ```
+ * @see Assertions.assertNull
+ */
+@API(since = "5.11", status = EXPERIMENTAL)
+fun assertNull(actual: Any?, message: String) {
+    contract {
+        returns() implies (actual == null)
+    }
+
+    Assertions.assertNull(actual, message)
+}
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: String? = ...
+ *
+ * assertNull(string) { "Should be nullable" }
+ *
+ * // compiler won't allow even safe calls, since the string is always null
+ * // string?.isNotEmpty()
+ * ```
+ * @see Assertions.assertNull
+ */
+@API(since = "5.11", status = EXPERIMENTAL)
+fun assertNull(actual: Any?, messageSupplier: () -> String) {
+    contract {
+        returns() implies (actual == null)
+
+        callsInPlace(messageSupplier, AT_MOST_ONCE)
+    }
+
+    Assertions.assertNull(actual, messageSupplier)
+}
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: String? = ...
+ *
+ * assertNotNull(string)
+ *
+ * // compiler smart casts nullableString to a non-nullable object
+ * assertTrue(string.isNotEmpty())
+ * ```
+ * @see Assertions.assertNotNull
+ */
+@API(since = "5.11", status = EXPERIMENTAL)
+fun assertNotNull(actual: Any?) {
+    contract {
+        returns() implies (actual != null)
+    }
+
+    Assertions.assertNotNull(actual)
+}
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: String? = ...
+ *
+ * assertNotNull(string, "Should be non-nullable")
+ *
+ * // compiler smart casts nullableString to a non-nullable object
+ * assertTrue(string.isNotEmpty())
+ * ```
+ * @see Assertions.assertNotNull
+ */
+@API(since = "5.11", status = EXPERIMENTAL)
+fun assertNotNull(actual: Any?, message: String) {
+    contract {
+        returns() implies (actual != null)
+    }
+
+    Assertions.assertNotNull(actual, message)
+}
+
+/**
+ * Example usage:
+ * ```kotlin
+ * val string: String? = ...
+ *
+ * assertNotNull(string) { "Should be non-nullable" }
+ *
+ * // compiler smart casts nullableString to a non-nullable object
```

Comment:
```suggestion
 * // compiler smart casts string to a non-nullable object
```

Reply:
The examples in the docs have been revised and variable names have been slightly adjusted